bootstrap/bungeecord/src/main/java/org/geysermc/geyser/platform/bungeecord/GeyserBungeePingPassthrough.java

```diff
@@ -25,164 +25,556 @@
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public ChannelFuture disconnect(ChannelPromise channelPromise) {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public ChannelFuture close(ChannelPromise channelPromise) {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public ChannelFuture deregister(ChannelPromise channelPromise) {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public ChannelHandlerContext read() {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public ChannelFuture write(Object o) {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public ChannelFuture write(Object o, ChannelPromise channelPromise) {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public ChannelHandlerContext flush() {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public ChannelFuture writeAndFlush(Object o, ChannelPromise channelPromise) {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public ChannelFuture writeAndFlush(Object o) {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public ChannelPromise newPromise() {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public ChannelProgressivePromise newProgressivePromise() {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public ChannelFuture newSucceededFuture() {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public ChannelFuture newFailedFuture(Throwable throwable) {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public ChannelPromise voidPromise() {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public ChannelPipeline pipeline() {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public ByteBufAllocator alloc() {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public <T> Attribute<T> attr(AttributeKey<T> attributeKey) {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public <T> boolean hasAttr(AttributeKey<T> attributeKey) {
+            throw new UnsupportedOperationException();
+        }
     }
 
-    private static class GeyserPendingConnection implements PendingConnection {
+    @AllArgsConstructor
+    private static class GeyserChannel implements Channel {
```

Comment:
Would you be able to extend EmbeddedChannel or AbstractChannel here instead to reduce boilerplate?

Reply:
EmbeddedChannel wouldn't work (will internally create a new event loop, queues, unsafe, close futures, etc.) and I just need to override a single method, and when BungeeCord changes its InitialHandler, it could lead to unexpected behavior, and therefore its better to just get an exception. Same as with AbstractChannel.